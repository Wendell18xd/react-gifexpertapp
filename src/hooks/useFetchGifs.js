import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

const useFetchGifs = (categories) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(categories)
            .then(img => {

                setState({
                    data: img,
                    loading: false
                })

            })
    }, [categories])

    return state;
}

export default useFetchGifs
