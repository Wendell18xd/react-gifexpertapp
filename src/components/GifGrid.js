import React from 'react'
import PropTypes from 'prop-types'
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ categories }) => {

    const { data, loading } = useFetchGifs(categories);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{categories}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    data.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }

            </div>
        </>
    )
}

GifGrid.propTypes = {
    categories: PropTypes.string.isRequired
}

export default GifGrid
