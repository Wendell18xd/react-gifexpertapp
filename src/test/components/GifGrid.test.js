import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs')

describe('Tests en componente Gifgrid', () => {

    const categories = "Goku"

    test('Debe cargar el componente GifGrid correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid categories={categories} />)
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://xxddxd.jpg',
            title: "cualquier cosa"
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid categories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })


})
