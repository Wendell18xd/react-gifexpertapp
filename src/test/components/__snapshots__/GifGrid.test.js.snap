// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en componente Gifgrid Debe cargar el componente GifGrid correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests en componente Gifgrid Debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://xxddxd.jpg"
    />
  </div>
</Fragment>
`;
